xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/delete/DocumentsDeletingListener.java

```diff
@@ -81,6 +81,12 @@ public DocumentsDeletingListener()
     @Override
     public void onEvent(Event event, Object source, Object data)
     {
+        CancelableEvent cancelableEvent = (CancelableEvent) event;
+        if (cancelableEvent.isCanceled()) {
+            logger.warn("Skipping " + this.getName() + " as the event is already cancelled.");
```

Comment:
```suggestion
            logger.warn("Skipping [{}] as the event is already cancelled.", this.getName());
```

Reply:
Ahah nice new Github feature but I'll apply it with the other changes ;)

Reply:
If it's normal than cancelled event stop propagation I wouldn't use a `warn` level but more a `debug` one.

Reply:
> If it's normal that cancelled events stop propagation

I made a proposal in that way, but It's not been accepted yet and Marius made some very good points against IMO. Now a warn might be a bit strong indeed. info could be appropriate here.

Reply:
Not info, it's more like debug if it's normal. Warn means it's not normal (meaning not normal that propagation is cancelled, I thought this is what we wanted and thus was normal). 

See https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HLoggingBestPractices